### PR TITLE
Add tests of fstatfs, epoll_create, epoll_ctl, and epoll_wait syscall

### PR DESCRIPTION
**Pull request by Yaxuan-w**
## Description

Add tests for `fstatfs`, `epoll_create`, `epoll_ctl`, and `epoll_wait` syscall. 

 

### Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Switch to the `safeposix-rust` directory and do `cargo clean`, then run `cargo test --lib`.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been added to a pull request and/or merged in other modules (native-client, lind-glibc, lind-project)

### CODE REVIEW

**Comment by yizhuoliang** [comment]
The fstatfs one is simple and looks good. The epoll one is lengthy yet understandable with the comments. If you still have time, I would recommend trying to register a socket fd that's already closed or doesn't exist, and see if the epoll really complains.

**Review by justyoyo123 — APPROVED** [review]
Both the fstatfs and epoll test are well commented and make sense. Looks good to me.

**Review by kuzeyardabulut — APPROVED** [review]
The comments are very nice and understandable. Looks good to me.

File: src/tests/pipe_tests.rs [hunk 1]
```diff
@@ -105,4 +105,6 @@ pub mod pipe_tests {
 
         lindrustfinalize();
     }
+
```
Comment (rennergade) [inline src/tests/pipe_tests.rs:108]: remove this extra added whitespace

**Review by rennergade — CHANGES_REQUESTED** [review]
Minor whitespace removal request. Otherwise I think this looks very good. Thanks!

**Pull request merged**